controller/generic/metacontroller.go

```diff
@@ -82,19 +84,15 @@ type ConfigMetaController struct {
 	// controllers
 	Configs []*v1alpha1.GenericController
 
-	// Total timeout for any condition to succeed.
-	//
-	// NOTE:
-	//	This is currently used to load config that is required
-	// to run Metac.
-	WaitTimeoutForCondition time.Duration
+	// This will allow executing start logic to be retried
+	// indefinitely till all the watch controllers are started
+	RetryIndefinitelyForStart *bool
```

Comment:
Maybe I need more clarification on that, maybe you can explain difference how it was working without this change to allow better  understanding of implications ?

Reply:
Yes. Sure. I have updated various code comments as well as commit & PR message.
Please review.